File: lib/commonlib/src/main/java/com/android/commonlib/utils/SPUtils.java
package com.android.commonlib.utils;

import android.content.Context;
import android.content.SharedPreferences;
import android.text.TextUtils;

import com.android.commonlib.base.BaseApplication;
import com.google.gson.Gson;


/**
 * 对Sharedpreferences功能的封装
 */
public class SPUtils {


    public static void putBoolean(String key, boolean value) {
        SharedPreferences sp = BaseApplication.getInstance().getSharedPreferences(Constants.SP_NAME, Context.MODE_PRIVATE);
        //添加保存数据
        sp.edit().putBoolean(key, value).apply();
    }

    public static boolean getBoolean(String key, boolean defValue) {
        SharedPreferences sp = BaseApplication.getInstance().getSharedPreferences(Constants.SP_NAME, Context.MODE_PRIVATE);
        return sp.getBoolean(key, defValue);
    }

    public static void putString(String key, String value) {
        SharedPreferences sp = BaseApplication.getInstance().getSharedPreferences(Constants.SP_NAME, Context.MODE_PRIVATE);
        //添加保存数据
        sp.edit().putString(key, value).apply();

    }

    public static String getString(String key, String defValue) {
        SharedPreferences sp = BaseApplication.getInstance().getSharedPreferences(Constants.SP_NAME, Context.MODE_PRIVATE);
        return sp.getString(key, defValue);

    }

    public static void putInt(String key, int value) {
        SharedPreferences sp = BaseApplication.getInstance().getSharedPreferences(Constants.SP_NAME, Context.MODE_PRIVATE);
        //添加保存数据
        sp.edit().putInt(key, value).apply();

    }

    public static int getInt(String key, int defValue) {
        SharedPreferences sp = BaseApplication.getInstance().getSharedPreferences(Constants.SP_NAME, Context.MODE_PRIVATE);
        return sp.getInt(key, defValue);

    }

    public static void putLong(String key, long value) {
        SharedPreferences sp = BaseApplication.getInstance().getSharedPreferences(Constants.SP_NAME, Context.MODE_PRIVATE);
        //添加保存数据
        sp.edit().putLong(key, value).apply();

    }

    public static long getLong(String key, Long defValue) {
        SharedPreferences sp = BaseApplication.getInstance().getSharedPreferences(Constants.SP_NAME, Context.MODE_PRIVATE);
        return sp.getLong(key, defValue);
    }

    /**
     * 添加对象
     *
     * @param key
     * @param t
     */
    public static <T> void putModel(String key, T t) {
        if (!TextUtils.isEmpty(key)) {
            Gson gson = new Gson();
            if (t != null) {
                putString(key, gson.toJson(t));
            } else {
                putString(key, "");
            }
        }
    }

    /**
     * 获取对象
     *
     * @param key
     * @param clazz
     * @return
     */
    public static <T> T getModel(String key, Class<T> clazz) {
        String value = null;
        if (!TextUtils.isEmpty(key)) {
            value = getString(key, "");
        }
        Gson gson = new Gson();
        return TextUtils.isEmpty(value) ? null : gson.fromJson(value, clazz);// fastjson
    }

}

File: autoclick/src/main/java/com/auto/click/utils/SpUtils.kt
package com.auto.click.utils

import android.content.Context
import android.content.SharedPreferences

/**
 * @Author Roger
 * @Date 2023/3/22 15:51
 * @Description
 */
object SpUtils {

    private var mSp: SharedPreferences? = null
    const val SP_NAME = "autoclick"

    fun init(context: Context) {
        mSp = context.getSharedPreferences(SP_NAME, Context.MODE_PRIVATE)
    }

    fun putString(key: String?, `val`: String?) {
        mSp!!.edit().putString(key, `val`).apply()
    }

    fun putStringCommit(key: String?, `val`: String?) {
        mSp!!.edit().putString(key, `val`).apply()
    }

    fun getString(key: String?, def: String?): String? {
        return mSp!!.getString(key, def)
    }

    fun getBoolean(key: String?, def: Boolean): Boolean {
        return mSp!!.getBoolean(key, def)
    }

    fun putBoolean(key: String?, `val`: Boolean) {
        mSp!!.edit().putBoolean(key, `val`).apply()
    }

    fun remove(key: String?) {
        mSp!!.edit().remove(key).apply()
    }

    fun getSp(): SharedPreferences? {
        return mSp
    }

    fun getLong(key: String?, `val`: Long): Long {
        return mSp!!.getLong(key, `val`)
    }

    fun putLong(key: String?, `val`: Long) {
        mSp!!.edit().putLong(key, `val`).apply()
    }

    fun getInt(key: String?, `val`: Int): Int {
        return mSp!!.getInt(key, `val`)
    }

    fun putInt(key: String?, `val`: Int) {
        mSp!!.edit().putInt(key, `val`).apply()
    }
}